scripts/west_commands/patch.py

```diff
@@ -338,26 +379,23 @@ def clean(self, args, yml, mods=None):
                 # If this fails for some reason, just log it and continue
                 self.err(f"failed to clean up {mod}: {e}")
 
-    def list(self, args, yml, mods=None):
+    def list(self, args, yml, dest_mods=None):
         patches = yml.get("patches", [])
         if not patches:
             return
 
         for patch_info in patches:
-            if mods and Path(patch_info["module"]) not in mods:
+            if dest_mods and patch_info["module"] not in dest_mods:
                 continue
             self.inf(patch_info)
 
-    @staticmethod
-    def get_mod_paths(args, yml):
-        patches = yml.get("patches", [])
-        if not patches:
-            return {}
-
-        mod_paths = {}
-        for patch_info in patches:
-            mod = Path(patch_info["module"])
-            mod_path = os.path.realpath(Path(args.west_workspace) / mod)
-            mod_paths[mod] = mod_path
+    def get_module(self, module):
+        all_modules = zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest)
+        if module == 'zephyr':
+            Module = namedtuple('Module', ['project'])
+            return Module(ZEPHYR_BASE)
+        for m in all_modules:
+            if m.meta['name'] == module:
+                return m
 
-        return mod_paths
+        return None
```

Comment:
This is the redundant complexity that I mentioned earlier.

We're forced to parse the main `west.yml` file from the `zephyr` module in order to transform `name` into `path` instead of just using `path` in the first place.

I'm also not entirely sure - does this include modules defined outside of the main `west.yml`? I.e. in T2 Star Topology?

Reply:
It includes what is configured as `manifest.path`/`manifest.file` in `west config` and everything that is imported from there.